### PR TITLE
Fixes pose publishing to include/exclude odom frame.

### PR DESCRIPTION
**Pull request by wohe**
Changes the pose publishing to always publish the odom to map transform.
Either expects or provides (configurable) the odom to tracking transform.
Also cleans up the RViz plugin code a bit.

### CODE REVIEW

File: cartographer_ros/src/cartographer_node_main.cc [hunk 1]
```diff
@@ -515,24 +519,41 @@ void Node::PublishSubmapList(int64 timestamp) {
...
+    stamped_transform.transform = ToGeometryMsgTransform(tracking_to_odom);
+    tf_broadcaster_.sendTransform(stamped_transform);
+  } else {
+    try {
+      const Rigid3d tracking_to_odom =
+          LookupToTrackingTransformOrThrow(time, odom_frame_).inverse();
+      const Rigid3d odom_to_map = tracking_to_map * tracking_to_odom.inverse();
+      stamped_transform.transform = ToGeometryMsgTransform(odom_to_map);
+      tf_broadcaster_.sendTransform(stamped_transform);
+    } catch (tf2::TransformException& ex) {
```
Comment (damonkohler) [inline cartographer_ros/src/cartographer_node_main.cc]: const

File: cartographer_ros/src/submaps_display.cc [hunk 2]
```diff
@@ -79,12 +79,12 @@ SubmapsDisplay::SubmapsDisplay()
       QString::fromStdString(
           ros::message_traits::datatype<::cartographer_ros_msgs::SubmapList>()),
       "cartographer_ros_msgs::SubmapList topic to subscribe to.", this,
-      SLOT(UpdateTopic()));
+      SLOT(UpdateSubmapTopicOrQuery()));
```
Comment (damonkohler) [inline cartographer_ros/src/submaps_display.cc]: UpdateSubmapTopicOrService

File: cartographer_ros/src/cartographer_node_main.cc (hunk 1 above)
Comment (wohe) [inline cartographer_ros/src/cartographer_node_main.cc]: Done.

File: cartographer_ros/src/submaps_display.cc (hunk 2 above)
Comment (wohe) [inline cartographer_ros/src/submaps_display.cc]: Done.

**Comment by damonkohler** [comment]
LGTM

**Pull request merged**